src/cpp/rtps/builtin/BuiltinProtocols.cpp

```diff
@@ -125,6 +125,11 @@ bool BuiltinProtocols::initBuiltinProtocols(
         mp_WLP->initWL(mp_participantImpl);
     }
 
+    if(m_att.discovery_config.discoveryProtocol == DiscoveryProtocol_t::SIMPLE)
```

Comment:
I think this should also be done for CLIENT. See my other comment

Reply:
Updated